Kibana Deployment

elipe17 opened the pull request:
## Summary of Changes
- Added manifests to deploy Kibana and the Elastic proxy
- Updated Nginx to act as an auth endpoint for Kibana
- Removed elastic_setup container because it is now irrelevant
- Updated deployment configs
Pull request closes #2790 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose up
cd tdrs-backend && docker-compose up
```

1. Open http://localhost:3000/ and sign in.
2. Navigate to /kibana and verify you get an error from nginx
3. Add BYPASS_KIBANA_AUTH=yes to your .env
4. Navigate to /kibana and verify you can see Kibana and it feels native to frontend
5. Navigate to https://tdp-frontend-raft.app.cloud.gov and perform the previous steps
  - To see Kibana in the deployed environment follow the steps below to give yourself access. Make sure you have already requested access by way of the frontend. If you already have access, the steps below still apply. If you don't have the cf cli tool, let me know and I can perform the steps for you after you have requested access from the frontend.
  1. Login to cloud foundry with the CLI: `cf login --sso`
  2. Ssh into backend: `cf ssh tdp-backend-raft`
  3. Enable python: `/tmp/lifecycle/shell`
  4. Start Django shell: `python manage.py shell_plus`
  5. Get your user: `user = User.objects.all().filter(email="YOUR_LOGIN_EMAIL_HERE")[0]`
  6. Approve your user: `user.account_approval_status = "Approved"`
  7. Get correct group: `group = Group.objects.get(name="OFA System Admin")`
  8. Add yourself to group: `group.user_set.add(user)`
  9. Give user a fake hhs_id: `user.hhs_id = "111111111111"`
  10. Save: `user.save()`
  11. Verify you can see the frontend now and the Kibana tab.

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x]   Kibana deploys to dev environment
- [x]  Routes between Kibana and Elastic are configured
- [x]  Kibana can connect to Elastic
- [x]  User can navigate to Kibana from frontend
- [x] Testing Checklist has been run and all tests pass
- [x] README is updated, if necessary
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?


### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

raftmsohani committed:
Added formating for header and autofit columns

raftmsohani committed:
Formatted the headers

raftmsohani committed:
added year/month to the columns

raftmsohani committed:
Added contants - translation column

raftmsohani committed:
added friendly names to T1 and T2

raftmsohani committed:
added friendly name to m1 and m2

raftmsohani committed:
added friendly name to m3

raftmsohani committed:
added friendly_name to t3

raftmsohani committed:
added friendly_name to t4 and t5

raftmsohani committed:
added friendly_name to t7

raftmsohani committed:
correct missing friendly_name

raftmsohani committed:
correction on failing tests

raftmsohani committed:
addedfriendly name to excel report

raftmsohani committed:
linting

raftmsohani committed:
linting

raftmsohani committed:
linting

raftmsohani committed:
delete contants.py

raftmsohani committed:
added test for json field in error model

raftmsohani committed:
linting

raftmsohani committed:
linting

raftmsohani committed:
linting

raftmsohani committed:
2599-added friendly name to postparsing validators

raftmsohani committed:
refining the validator tests

raftmsohani committed:
added returning fields names to validators

raftmsohani committed:
added friendly_name to error field

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
linting

raftmsohani committed:
corrections on views/tests

raftmsohani committed:
corrections for fields

raftmsohani committed:
failing test corrected

raftmsohani committed:
failing test corrected

raftmsohani committed:
correcting test failures

raftmsohani committed:
linting

raftmsohani committed:
corrected the excel fiel generator

raftmsohani committed:
removed excessive space in validator

raftmsohani committed:
linting

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
listing

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
added m6

raftmsohani committed:
lint

raftmsohani committed:
corrected new line break

raftmsohani committed:
refactored validator logic

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
linting and correction on t1

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
friendly_name correction from comments

raftmsohani committed:
Merge branch '2599-error-report-corrections-impact-wpr-metrics' of https://github.com/raft-tech/TANF-app into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
friendly_name correction

raftmsohani committed:
corrected failing test for m5

raftmsohani committed:
refactor the field_json creation DRY

Commit:
Changes for fully local development
 - Enables direct frontend/backend communication sans Login.gov/Cloud.gov
 - Drives off new DEVELOPMENT env var
 - Pre-configures and disables frontend auth functionality
 - Testing based on new dev user
   - Install via web: ./manage.py generate_dev_user

elipe17 committed:
- Added Kibana config

raftmsohani committed:
friendly_name corrections

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

elipe17 committed:
Merge branch '2599-error-report-corrections-impact-wpr-metrics' of https://github.com/raft-tech/TANF-app into kibana-local

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into kibana-local

raftmsohani committed:
linting and cleaning errors

raftmsohani committed:
linting

raftmsohani committed:
correction on friendly_names

raftmsohani committed:
corrected friendly_name for test_util

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

raftmsohani committed:
correction child care - number of months

raftmsohani committed:
Merge branch '2599-error-report-corrections-impact-wpr-metrics' of https://github.com/raft-tech/TANF-app into 2599-error-report-corrections-impact-wpr-metrics

Commit:
Reorganized front end logic on REACT_APP_DEVAUTH env var

Commit:
Reorganized backend logic on REACT_APP_DEVAUTH env var

George-Hudson committed:
fixed a few more typos and some spacing. (#2767)

* fixed a few more typos and some spacing.

* fixed linting issues

* missed a spot.

---------

Co-authored-by: George Hudson <ghudson@teamraft.com>

elipe17 committed:
- Added basic security to Kibana/Elastic
- Added setup container to init elastic users, roles, and passwords

elipe17 committed:
- Remove debug code

elipe17 committed:
- change provider name

elipe17 committed:
- Updating settings to reference environment variables

raftmsohani committed:
Merge branch 'develop' into 2599-error-report-corrections-impact-wpr-metrics

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
- Add elastic dependency

elipe17 committed:
- Fix network issue

elipe17 committed:
- Added bulk creation of elastic indices

elipe17 committed:
- Updated schemas to reference model based off of elastic document

elipe17 committed:
- Remove password auth from elastic/kibana

elipe17 committed:
- Remove password auth

elipe17 committed:
- Fix tests

elipe17 committed:
- Fix lint

elipe17 committed:
- remove debug print

elipe17 committed:
Merge branch '2599-error-report-corrections-impact-wpr-metrics' of https://github.com/raft-tech/TANF-app into elastic-bulk-doc-creation

elipe17 committed:
Merge branch '1858-kibana-ofa-access' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
- Added proof on concept for tdp based kibana auth

elipe17 committed:
- Fixing type issue

Commit:
added is_superuser and is_staff attrs to dev user

elipe17 committed:
- Add group check

elipe17 committed:
- Add frontend group check for kibana

elipe17 committed:
- fix lint

elipe17 committed:
- Fix lint errors

elipe17 committed:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
- Fix doc strings

elipe17 committed:
- Adding authenticated permission

elipe17 committed:
- Renaming variables to clarify things

elipe17 committed:
- fix lint

elipe17 committed:
Revert "- Remove password auth from elastic/kibana"

This reverts commit 522ca381651e005e35701d51b94e6cf4d78c011a.

elipe17 committed:
- Setting up anonymous users with kibana_admin privileges

elipe17 committed:
- Adding password to settings in cloud.gov

elipe17 committed:
- remove incorrect auth
- use admin only in frontend and backend

elipe17 committed:
- Add elastic profile

Commit:
DevAuth feature redesign inspired by Cypress
 - Initializing frontend w/POST /login/cypress: {devEmail, local-cypress-token}
 - Changed REACT_APP_DEVAUTH to provide the email of the desired dev user
 - Modified CustomAuthentication.authenticate to handle both known use cases
 - Added stt_id=31 to the initial dev user
 - Disabled ES disk threshold checking for local dev which blocked ES startup
 - Removed DevAuthentication and other now unnecessary code

Commit:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into elastic-bulk-doc-creation
Merging latest changes from server

Commit:
Fixed CustomAuthentication.authenticate return val for login.py use case

Commit:
Fixed CustomAuthentication.authenticate logging for login.py use case

Commit:
Removed unneeded permissions import

Commit:
Updates to REACT_APP_DEVAUTH env var settings
 - Enabled with an email address value
 - Disabled by default

elipe17 committed:
- debugging env vars

elipe17 committed:
- Testing what settings are used

elipe17 committed:
Revert "- debugging env vars"

This reverts commit 900efa879e8cb33f0c62140ed4d8e9a1e1b5496a.

elipe17 committed:
Revert "- Testing what settings are used"

This reverts commit 784530e49d584db5bda89714a7de4f56c3ee805e.

elipe17 committed:
- debugging env vars again

elipe17 committed:
- Switching to container networking

Commit:
Restored support for CustomAuthentication.authenticate username keyword

Commit:
Modified CustomAuthentication.authenticate comment to satisfy flake8

elipe17 committed:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into elastic-bulk-doc-creation

elipe17 committed:
commit

elipe17 committed:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
asdfgvasd

elipe17 committed:
Revert "Modified CustomAuthentication.authenticate comment to satisfy flake8"

This reverts commit 761e4eb253d366ef742dd8caf94b6220ed9e81a1.

elipe17 committed:
Revert "Restored support for CustomAuthentication.authenticate username keyword"

This reverts commit 4bf895722e356e79b8bbe3674361b90888b91752.

elipe17 committed:
Revert "Updates to REACT_APP_DEVAUTH env var settings"

This reverts commit 7fc2a09353804fb728852e9accc041dff08e44e3.

elipe17 committed:
Revert "Removed unneeded permissions import"

This reverts commit c18383fe2bf8352c50dd84d2a84408fef2b71367.

elipe17 committed:
Revert "Fixed CustomAuthentication.authenticate logging for login.py use case"

This reverts commit 2b9b46f0e719638b320d0a7bbb2bd87eda97eeef.

elipe17 committed:
Revert "Fixed CustomAuthentication.authenticate return val for login.py use case"

This reverts commit 97a0cf6995dc17937f083d5efde1d74866c01ff4.

elipe17 committed:
Revert "DevAuth feature redesign inspired by Cypress"

This reverts commit 1497d4ab7549bf674e1f71d8f8f039ec7de363bf.

elipe17 committed:
Revert "commit"

This reverts commit a284856c66cb2caaca4b471c2f81fc39421e8f00.

elipe17 committed:
Revert "added is_superuser and is_staff attrs to dev user"

This reverts commit 6ffbee8f588f12b7595abd7adcfe15ad7e70d11b.

elipe17 committed:
Revert "Reorganized backend logic on REACT_APP_DEVAUTH env var"

This reverts commit 7fd7b4d48cd30a7e2c142ea7f1d85f8df95b80d8.

elipe17 committed:
Revert "Reorganized front end logic on REACT_APP_DEVAUTH env var"

This reverts commit 32a46713ae102fa15fdab0fcb66fca99b42eb7e2.

elipe17 committed:
Revert "Changes for fully local development"

This reverts commit 556221b310b73bee5e9af32eb4cd603b61a25d02.

elipe17 committed:
asdf

elipe17 committed:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
- Adding integration tests for elastic bulk doc creation

elipe17 committed:
Revert "asdf"

This reverts commit 26455b48582ca9c6d986377e56475644525f7665.

elipe17 committed:
- fix lint

elipe17 committed:
fasdf

elipe17 committed:
- Failed buildpack deploy. Commiting for history

elipe17 committed:
- Updating manifests to deploy proxy and kibana

elipe17 committed:
- Adding working manifests for kibana and proxy

elipe17 committed:
- moving manifest to its own directory

elipe17 committed:
- Updating backend deployment to include kibana and proxy for circi deploys

elipe17 committed:
- remove port

elipe17 committed:
- allowing manifest

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 committed:
- Update kibana and proxy hostnames

elipe17 committed:
- adding debug

elipe17 committed:
- Updating schemas

elipe17 committed:
- Fix mispelling

elipe17 committed:
- inplace update

elipe17 committed:
- Fixing var names

elipe17 committed:
- remove unset
- Add set-env for proxy

elipe17 committed:
- parametrizing proxy host name for kibana

elipe17 committed:
- Adding debug logging to see whats up

elipe17 committed:
- fix lint

elipe17 committed:
- adding kibana to deployed nginx conf

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into elastic-bulk-doc-creation

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
Merge branch '1858-kibana-ofa-access' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 committed:
- Added usage of document to tribal

elipe17 committed:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
Merge branch '1858-kibana-ofa-access' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 committed:
- making url public for the time being to allow redirect

elipe17 committed:
- testing 2GB again

elipe17 committed:
- making stt searchable
- update proxy mem limits

elipe17 committed:
- back to internal route

elipe17 committed:
- pushing temp changes for now

elipe17 committed:
- adding extra setting

andrew-jameson committed:
Merge branch 'develop' into elastic-bulk-doc-creation

elipe17 committed:
- pushing changes

elipe17 committed:
- removing secondary proxy

elipe17 committed:
- nginx auth

elipe17 committed:
- Updated to allow nginx auth proxy for kibana

elipe17 committed:
- adding back headers

elipe17 committed:
- Undoing temp changes

elipe17 committed:
- Updating to support cloud.gov deploy

elipe17 committed:
- Fixing warnings

elipe17 committed:
- fix env var

elipe17 committed:
- Add netpol to allow kibana to talk to frontend

elipe17 committed:
- Adding env vars for kibana

elipe17 committed:
- fixing env var name

elipe17 committed:
- remove host as test
- remove invalid params from search indices

elipe17 committed:
- remover server host param

elipe17 committed:
- remove request limiter

elipe17 committed:
- Adding unsafe-inline

elipe17 committed:
- Updating to use path based env vars

elipe17 committed:
- whitelisting kibana in CSP

elipe17 committed:
- converting back to env vars only

elipe17 committed:
- adding unsafe eval

elipe17 committed:
- Updated based on feedback

elipe17 committed:
- Fixing error

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into elastic-bulk-doc-creation

elipe17 committed:
- Updating local docker image to match deployed image

elipe17 committed:
Merge branch 'elastic-bulk-doc-creation' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
Merge branch '1858-kibana-ofa-access' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 committed:
- Removing elastic setup as it is irrelevant now

raftmsohani commented on tdrs-frontend/nginx/cloud.gov/locations.conf:
```diff
@@ -22,6 +22,44 @@ location ~ ^/(v1|admin|static/admin|swagger|redocs) {
+    proxy_read_timeout    300;
+    proxy_send_timeout    300;
+    send_timeout 	      900;
+    proxy_buffer_size     4k;
```
I don't think we need buffer for auth, also not the " proxy_temp_file_write_size"

elipe17 committed:
- Updating frontend to only allow access to kibana sitemap if the user is Dev or Sys Admin

raftmsohani reviewed (approved):
LGTM! great work 👍

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 1858-kibana-ofa-access

elipe17 committed:
- fix lint

elipe17 committed:
Merge branch '1858-kibana-ofa-access' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 committed:
- remove unnecessary nginx settings

jtimpe reviewed (approved):
Excellent work! 🚀

ADPennington commented on tdrs-frontend/nginx/cloud.gov/buildpack.nginx.conf:
```diff
@@ -61,9 +61,9 @@ http {
-        set $CSP "${CSP}script-src 'self';";
-        set $CSP "${CSP}script-src-elem 'self';";
-        set $CSP "${CSP}script-src-attr 'self' 'unsafe-inline';";
+        set $CSP "${CSP}script-src 'self' 'unsafe-eval' 'unsafe-inline' http://{{env "KIBANA_BASE_URL"}}:5601;";
```
fyi:  this unsafe-inline is showing up in security scans and if we need this setting, we should document why. the only documentation im aware of on this lives [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/Technical-Documentation/Architecture-Decision-Record/016-zap-waiver.md), and im not sure if the justification is still accurate.

ADPennington replied:
related to the above: it stops after the endpoint page loads.

elipe17 replied:
Unfortunately this is a requirement with Kibana and it likely will be for a long time. Kibana uses libraries that require the `unsafe-*`policies. The version of Kibana we are using was released in [October, 2019](https://www.elastic.co/blog/kibana-7-4-0-released) and [THIS TOPIC](https://discuss.elastic.co/t/does-kibana-need-the-autorisation-of-unsafe-inline-or-unsafe-eval-to-work-properly/234390#:~:text=May%202020-,Hey,-%40Otmane_Faouzi) states that as of May, 2020 the the requirement of these policies still exists. [THIS ISSUE](https://github.com/elastic/kibana/issues/132595l) captures the work that would allow the removal of those policies. However, it was opened in 2022 and is still open. 

Would you like me to update the documentation you linked, or should I create a new doc capturing this information?

ADPennington replied:
@elipe17 I think this is worth addressing as part of #2843.

ADPennington commented on tdrs-backend/tdpservice/users/api/authorization_check.py:
```diff
@@ -66,6 +66,8 @@ def get(self, request, *args, **kwargs):
 
         if (user.hhs_id is not None and user_in_valid_group) or settings.BYPASS_KIBANA_AUTH:
-            return HttpResponseRedirect(settings.KIBANA_BASE_URL)
+            logger.debug(f"User: {user} has correct authentication credentials. Allowing access to Kibana.")
```
noticing this line of code trigger 8-10x in the logs after clicking kibana endpoint.

elipe17 replied:
This log line will be hit every time the frontend makes a request to Kibana. This is because we need to pass through the authentication view every time prior to proxying the request to Kibana to ensure the user is still passing the traditional Auth Check and their group hasnt changed. Debug is the highest level for the logger so it wont show up in prod.

ADPennington reviewed (commented):
@elipe17 this looks good. Some summary-level notes/comments below. My testing notes are captured in more detail here: https://hackmd.io/P4sQkTRORLGjiodfgRwMtA?both#Kibana-Deployment

* If this will introduce new apps and resources (e.g. services) into the prod environment, we need the system diagrams and boundary diagram updated. This can be done in a follow-up ticket when this feature is more refined. 
* The Kibana tab is visible (the endpoint is not accessible) for `Developer` and `OFA System Admin` roles when signing in via login.gov. 
* We will still need to do some additional testing in staging to better understand what is causing the kibana routing issue.  
* I need more information on why we're using `unsafe-inline` and `unsafe-eval` settings since this often comes up in the security scans and we need to explain why to security team. 
* conflicts need to be resolved herein before merge.

elipe17 committed:
- Updated Kibana tab/link to only display when user is HHS AMS authenticated
- Added environment variable to show the tab/link for dev purposes

elipe17 committed:
Merge branch 'develop' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 commented:
@ADPennington I updated based on your feedback/responded to your questions. I updated the frontend to only display the Kibana tab if you are authenticated via AMS along with the existing controls. I also added a switch for local dev to display it since we cannot authenticate that way.

elipe17 committed:
- Commenting env var to default to correct behavior in any environment

elipe17 committed:
Merge branch 'develop' into 2790-deploy-kibana

elipe17 committed:
- update frontend memory quota

elipe17 committed:
Merge branch '2790-deploy-kibana' of https://github.com/raft-tech/TANF-app into 2790-deploy-kibana

elipe17 committed:
- Add resolver to nginx to avoid ip address change on app restart

elipe17 committed:
- Add dns fix for clamav

ADPennington commented on tdrs-frontend/nginx/cloud.gov/buildpack.nginx.conf:
```diff
@@ -34,6 +34,8 @@ http {
     limit_req_zone $binary_remote_addr zone=limitreqsbyaddr:20m rate=1000r/s;
     limit_req_status 444;
 
+    resolver {{nameservers}} valid=10s;
```
@elipe17 just noting for your reference that after adding missing env vars +  re-staging backend, i still had to restage frontend to authenticate. im not sure if this resolves the issue described in #2839

ADPennington commented on tdrs-frontend/src/selectors/auth.js:
```diff
@@ -62,4 +62,6 @@ export const accountCanViewAdmin = (state) =>
 
 export const accountCanViewKibana = (state) =>
   accountStatusIsApproved(state) &&
+  (selectUser(state)?.email?.includes('@acf.hhs.gov') ||
```
not sure its worth noting as an inline comment here that the developer role + acf email yields the kibana tabs but endpoint is still inaccessible. this might only be relevant to dev team and qasp reviewers since developer role doesnt exist in prod. @elipe17

ADPennington replied:
![2805p3](https://github.com/raft-tech/TANF-app/assets/63075587/0d8ea1b6-c3da-4e78-b2c7-ee601d6f5b5d)

elipe17 replied:
Good catch, I will remove developer!

elipe17 committed:
- OFA sys admin only

elipe17 committed:
- Testing dns resolution

elipe17 committed:
- Fix merge conflict that was causing incorrect routing to kibana locally

elipe17 committed:
- Updated to be boolean

elipe17 committed:
- Remove merge conflict

elipe17 committed:
- fix var

elipe17 committed:
Revert "- Testing dns resolution"

This reverts commit 84aac744fd999752d3dea8c91c9dcf3b9b3173bc.

elipe17 committed:
- removing unnecessary setting

elipe17 committed:
- Renaming var to capture its use better

elipe17 committed:
- Keeping var commented out

ADPennington reviewed (approved):
excellent work @elipe17 thanks for all the testing support! 🚀 

- added system diagram updates as a task to #1441 
- #2839 will capture work for troubleshooting routing issue
- #2843 will capture updates to security tool ADR re: our use of unsafe-* policies
